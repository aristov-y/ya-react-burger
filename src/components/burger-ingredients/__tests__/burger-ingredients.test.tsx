import React from 'react';
import { create, act, ReactTestRenderer } from 'react-test-renderer';
import BurgerIngredients from '../burger-ingredients';
import { useDispatch, useSelector } from 'react-redux';

jest.mock('../burger-ingredients-section', () => () => 'BurgerIngredientsSection');
const store = {
  ingredients: {
    ingredients: []
  }
}

const dispatch = jest.fn();

beforeEach(() => {
  (useSelector as jest.Mock).mockImplementation((fn) => fn(store));
  (useDispatch as jest.Mock).mockImplementation(() => dispatch);
});

describe('burger-ingredients', function () {
  it('should render', function () {
    let component: ReactTestRenderer;
    act(() => {
      component = create(<BurgerIngredients className={"test-class"} />);
    });

    expect(component.toJSON()).toMatchSnapshot();
  });
});
