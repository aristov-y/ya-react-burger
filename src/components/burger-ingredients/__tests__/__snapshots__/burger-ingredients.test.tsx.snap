// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`burger-ingredients should render 1`] = `
<div
  className="test-class burger-ingredients mt-10"
>
  <span
    className="text text_type_main-medium"
  >
    Соберите бургер
  </span>
  <div
    className="mt-5"
    style={
      Object {
        "display": "flex",
      }
    }
  >
    Tab
    Tab
    Tab
  </div>
  <div
    className="mt-10 burger-ingredients-scroller"
  >
    BurgerIngredientsSection
    BurgerIngredientsSection
    BurgerIngredientsSection
  </div>
</div>
`;
